src/skin/qml/qmlplayerproxy.cpp

```diff
@@ -0,0 +1,129 @@
+#include "skin/qml/qmlplayerproxy.h"
+
+#include "mixer/basetrackplayer.h"
+
+// FIXME: Unfornately, clazy can stumble over these preprocessor macros and
+// show false-positive warnings. Therefore, the clazy-incorrect-emit warnings
+// are disabled in this source file.
+//
+// clazy:excludeall=incorrect-emit
+
+#define PROPERTY_IMPL(TYPE, NAME, GETTER, SETTER)    \
+    TYPE QmlPlayerProxy::GETTER() const {            \
+        const TrackPointer pTrack = m_pCurrentTrack; \
+        if (pTrack == nullptr) {                     \
+            return TYPE();                           \
+        }                                            \
+        return pTrack->GETTER();                     \
+    }                                                \
+                                                     \
+    void QmlPlayerProxy::SETTER(const TYPE& value) { \
+        const TrackPointer pTrack = m_pCurrentTrack; \
+        if (pTrack != nullptr) {                     \
+            pTrack->SETTER(value);                   \
+            emit NAME##Changed();                    \
```

Comment:
Why signal here, rather than subscribe the track signal?

Reply:
the track doesn't expose signals for each property.

Reply:
Yes. You set the track property, the track checks if it has changed and emits the changed() signal.
this is cached in slotTrackChanged() and will emit the individual signal per property. 
So this one here should be redundant.

Reply:
I'd rather do the opposite and only emit the changed signal for poperties that actually changed. I'll look into adapting the track object.

Reply:
Should I rebase on #3924 or do we want to go with a more iterative approach?